modules/ServerAPI/src/client/Protocol.coffee

```diff
@@ -1127,10 +1184,59 @@ class EndpointListController extends AbstractFormController
 				@serviceReturn = null
 			dataType: 'json'
 
-	getCurrentEndpoints: => 
-		# get the current endpoints and their values for the protocol from @collection
+	getCurrentEndpoint: (lsState) => 
+		# gets the endpoint values for a single state
+		lsValues = lsState.attributes.lsValues.models
+
+		# create NAs for each entry in case we don't find a variable, we'll plug these in instead
+		endpointName = "NA"
+		endpointUnits = "NA"
+		endpointDataType = "NA"
+		endpointConc = "NA"
+		endpointConcUnits = "NA"
+		endpointTime = "NA"
+		endpointTimeUnits = "NA"
+		endpointHidden = "NA"
+
+		# extract the valid endpoint values from each lsState 
+		for lsValue in lsValues
+			lsValue = lsValue.attributes 
+
+			if lsValue.lsKind == "column name" and lsValue.ignored == false and lsValue.codeValue != undefined
+				endpointName = lsValue.codeValue 	
+			if lsValue.lsKind == "column units" and lsValue.ignored == false and lsValue.codeValue != undefined
+				endpointUnits = lsValue.codeValue
+			if lsValue.lsKind == "column type" and lsValue.ignored == false and lsValue.codeValue != undefined
+				endpointDataType = lsValue.codeValue
+			if lsValue.lsKind == "column concentration" and lsValue.ignored == false and lsValue.numericValue != undefined
+				endpointConc = lsValue.numericValue
+			if lsValue.lsKind == "column conc units" and lsValue.ignored == false and lsValue.codeValue != undefined
+				endpointConcUnits = lsValue.codeValue
+			if lsValue.lsKind == "column time" and lsValue.ignored == false and lsValue.numericValue != undefined
+				endpointTime = lsValue.numericValue
+			if lsValue.lsKind == "column time units" and lsValue.ignored == false and lsValue.codeValue != undefined
+				endpointTimeUnits = lsValue.codeValue
+			if lsValue.lsTypeAndKind == "codeValue_hide column" and lsValue.ignored == false and lsValue.codeValue != undefined
+				endpointHidden = lsValue.codeValue
+			
+			endpointStr = endpointName + endpointUnits + endpointDataType + String(endpointConc) + endpointConcUnits + String(endpointTime) + endpointTimeUnits + endpointHidden
+		
+		endpointData = 
```

Comment:
nit: I guess this works without an explicit `return` but I personally think it'd be clearer if this function had a 
`return endpointData` at the bottom

Reply:
That wasn't intentional! Weird that it works without it...